### PR TITLE
More cleanup of Terraform Tests

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

Just found a few small pieces still floating around that I missed when removing the Terraform tests.

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

N/A

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/kubernetes_update.md [hunk 1]
```diff
@@ -22,7 +22,6 @@ List of items to do for upgrading to {version_1} {version_2} {version_3}
     - [ ] Minikube in `build/includes/minikube.mk` (Get the patch version [here](https://kubernetes.io/releases/) since minikube supports the latest Kubernetes release)
     - [ ] Kind in `build/includes/kind.mk` (Confirm {version_2} is supported and get the patch version [here](https://github.com/kubernetes-sigs/kind/releases))
 - [ ] Update the k8s image used in the helm [pre-delete-hook](https://github.com/googleforgames/agones/blob/main/install/helm/agones/templates/hooks/pre_delete_hook.yaml) to {version_2} (Get the patch version [here](https://hub.docker.com/r/lachlanevenson/k8s-kubectl))
-- [ ] Update client-go in `go.mod` and `test/terraform/go.mod` to {version_2} by running `go get k8s.io/client-go@{CORRESPONDING_VERSION}` and `go get k8s.io/apiextensions-apiserver@{CORRESPONDING_VERSION}`, then re-run `go mod tidy` and `go mod vendor`
```
Comment (gongmax) [inline .github/ISSUE_TEMPLATE/kubernetes_update.md:25]: we still need this step for `go.mod`, right?

Comment (markmandel) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md:25]: Ah! Good catch. Will update, rather than delete 👍🏻

Comment (markmandel) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md:25]: This should be better now, and ready for re-review.

**Review by gongmax — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/kubernetes_update.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ List of items to do for upgrading to {version_1} {version_2} {version_3}
     - [ ] Minikube in `build/includes/minikube.mk` (Get the patch version [here](https://kubernetes.io/releases/) since minikube supports the latest Kubernetes release)
     - [ ] Kind in `build/includes/kind.mk` (Confirm {version_2} is supported and get the patch version [here](https://github.com/kubernetes-sigs/kind/releases))
 - [ ] Update the k8s image used in the helm [pre-delete-hook](https://github.com/googleforgames/agones/blob/main/install/helm/agones/templates/hooks/pre_delete_hook.yaml) to {version_2} (Get the patch version [here](https://hub.docker.com/r/lachlanevenson/k8s-kubectl))
```
Comment (Kalaiselvi84) [inline .github/ISSUE_TEMPLATE/kubernetes_update.md:24]: We need to update the link to:  https://hub.docker.com/layers/bitnami/kubectl/1.27.7/images/sha256-07726b35e759e4ac361ccca923c925959876e26b988e597528b48b65ef6d17f1?context=explore 

We use the `bitnami/kubectl` image.

Comment (Kalaiselvi84) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md:24]: May I check out your branch and update it? I've never tried this before.

Comment (markmandel) [reply .github/ISSUE_TEMPLATE/kubernetes_update.md]: Good catch - already done!

**Comment by markmandel** [comment]
That's a new flake - generic-1.28

```
time="2023-10-27 01:55:47.903" level=info msg="Finished Allocation."
    gameserverallocation_test.go:1190: 
        	Error Trace:	/go/src/agones.dev/agones/test/e2e/gameserverallocation_test.go:1190
        	Error:      	Not equal: 
        	            	expected: 100
        	            	actual  : 99
        	Test:       	TestGameServerAllocationDuringMultipleAllocationClients

```

**Pull request merged**